onmt/utils/loss.py

```diff
@@ -261,6 +261,51 @@ def _bottle(self, _v):
     def _unbottle(self, _v, batch_size):
         return _v.view(-1, batch_size, _v.size(1))
 
+    def ignore_prompt(
+        self, batch, response_token="▁Response", response_right_pattern_length=3
+    ):
+        """
+        Mask the prompt in the target side of the bath examples in order
+            to set the loss of the prompt to zero.
+        For finetuning on specific tasks.
+        The masks are supposed to be properly handled by the loss criterion
+            (e.g. nn.CrossEntropyLoss )
+
+        Args:
+        batch: The current batch.
+        response_token (str): The token used the locate the end of the prompt
+        response_right_pattern_length (int). The number of tokens in the response
+            pattern after the response_token
+
+        Ex: The response pattern is '### Response : ｟newline｠',
+            the response_token is '▁Response'.
+            and the tokenized response pattern is '##', '#', '▁Response', '▁:', '▁', '<0x0A>'].
+            Then response_right_pattern_length = 3
+        """
+        import numpy as np
```

Comment:
Can't we make it work with torch since it's already imported and used for the rest?